src/deposit.cairo

```diff
@@ -235,28 +301,18 @@ mod Deposit {
                 } else {
                     delta.amount1.mag.into()
                 };
-
-                if debt.into() > amount_swapped {
+                if debt > amount_swapped {
                     repaid_amount += amount_swapped;
                     debt_dispatcher.approve(zk_market.contract_address, amount_swapped.into());
                     zk_market.repay(debt_token, amount_swapped.try_into().unwrap());
                 } else {
-                    repaid_amount += debt.into();
-                    debt_dispatcher.approve(zk_market.contract_address, debt.into());
+                    repaid_amount += debt;
+                    debt_dispatcher.approve(zk_market.contract_address, debt);
                     zk_market.repay_all(debt_token);
                 }
 
-                debt = zk_market.get_user_debt_for_token(contract_address, debt_token);
-            };
-
-            let left = debt_dispatcher.balanceOf(contract_address);
-            let params = SwapParameters {
-                amount: i129 { mag: left.try_into().unwrap(), sign: false },
-                is_token1: !is_token1_repay_swap,
-                sqrt_ratio_limit: SQRT_LIMIT_WITHDRAW,
-                skip_ahead: 0
+                debt = zk_market.get_user_debt_for_token(contract_address, debt_token).into();
```

Comment:
If you previously called repay_all (line 311), this will be zero, no? Either remove or assert that it's zero in that case.